gatsby-node.ts

```diff
@@ -0,0 +1,177 @@
+import { GatsbyNode } from "gatsby"
+import path from "path"
+import { createFilePath } from "gatsby-source-filesystem"
+import { parse } from "svgson"
+import { pathThatSvg } from "path-that-svg"
+import fs from "fs"
+
+export const createPages: GatsbyNode["createPages"] = async ({
+  actions,
+  graphql,
+  reporter,
+}) => {
+  const { createPage } = actions
+
+  // Filter only for md files in the src/pages directory. Any md files found
+  // outside of that should not have a page created for it
+  const result = await graphql(`
+    {
+      allMarkdownRemark(filter: { fileAbsolutePath: { regex: "/(pages)/" } }) {
+        edges {
+          node {
+            id
+            fields {
+              slug
+            }
+            frontmatter {
+              template
+            }
+          }
+        }
+      }
+    }
+  `)
+
+  // Handle errors
+  if (result.errors) {
+    reporter.panicOnBuild(`Error while running GraphQL query.`)
+    return
+  }
+
+  // @ts-ignore
+  result.data.allMarkdownRemark.edges.forEach(({ node }) => {
+    const templateName = node.frontmatter.template || `default`
+    const template = path.resolve(
+      path.join("src/templates/", `${templateName}.tsx`)
```

Comment:
Is it possible to support `src/templates/${templateName}/index.tsx` here as well? In many cases I think our templates will be directories.